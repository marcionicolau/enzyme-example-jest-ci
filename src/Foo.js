// @flow

import React, { Component } from "react";
import PropTypes from "prop-types";

const propTypes = {};

const defaultProps = {};

class Foo extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return <div className="foo">Bar</div>;
  }
}

Foo.propTypes = propTypes;
Foo.defaultProps = defaultProps;

export default Foo;
