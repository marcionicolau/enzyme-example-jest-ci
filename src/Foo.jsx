// @flow

import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Foo extends Component {
  static propTypes = {
    loading: PropTypes.bool.isRequired,
    info: PropTypes.string,
  };

  constructor(props) {
    super(props);

    this.state = {
      info: props.info || '',
      loading: props.loading,
    };
  }

  render() {
    return <div className={this.state.loading ? 'foo-loading' : 'foo'}>{this.state.info}</div>;
  }
}

export default Foo;
