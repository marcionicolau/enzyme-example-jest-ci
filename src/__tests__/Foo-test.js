import React from "react";
import toJson from "enzyme-to-json";
import { shallow, mount, render } from "enzyme";

import Foo from "../Foo";

describe("A suite", function() {
  it("should render without throwing an error", function() {
    const info = "Bar";
    expect(
      shallow(<Foo loading info={info} />).contains(
        <div className="foo">{info}</div>
      )
    ).toBe(true);
  });

  it('should be selectable by class "foo"', function() {
    expect(shallow(<Foo loading />).is(".foo")).toBe(true);
  });

  it("should mount in a full DOM", function() {
    expect(mount(<Foo loading />).find(".foo").length).toBe(1);
  });

  it("should render to static HTML", function() {
    const info = "Bar";
    expect(render(<Foo loading info={info} />).text()).toEqual(info);
  });

  it("snapshot check", () => {
    const wrapper = shallow(<Foo loading />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
