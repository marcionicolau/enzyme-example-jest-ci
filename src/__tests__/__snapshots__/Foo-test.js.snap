// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite snapshot check 1`] = `
<div
  className="foo"
/>
`;
